core/commands/p2p.go

```diff
@@ -196,6 +202,40 @@ Example:
 	},
 }
 
+// checkPort checks whether target multiaddr contains tcp or udp protocol
+// and whether the port is equal to 0
+func checkPort(target ma.Multiaddr) error {
+	// get tcp or udp port from multiaddr
+	getPort := func() (string, error) {
```

Comment:
why not just a simple switch?

Reply:
However, We have no idea whether target multiaddr contains tcp or udp protocol.